### PR TITLE
bsp/u-boot: Replace remove usb power by Remove ADC

### PR DESCRIPTION
**Pull request by jorisoffouga**
Upstream U-Boot disable check_usb_power when CONFIG_ADC is disabled
see https://gitlab.denx.de/u-boot/u-boot/commit/415424737128bd2d17ca3a91f847a106aa615d67

Signed-off-by: Joris Offouga <offougajoris@gmail.com>

### CODE REVIEW

File: recipes-bsp/u-boot/u-boot-stm-common.inc [hunk 1]
```diff
@@ -8,7 +8,7 @@ DEPENDS += "bison-native"
 
 SRC_URI = " \
 	git://gitlab.denx.de/u-boot/u-boot.git;protocol=https \
-	file://0001-remove-usb-power.patch \
+	0001-configs-stm32mp1-Remove-ADC.patch \
```
Comment (texierp) [inline recipes-bsp/u-boot/u-boot-stm-common.inc]: file:// is missing :/

File: recipes-bsp/u-boot/u-boot-mainline/0001-configs-stm32mp1-Remove-ADC.patch [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+@@ -99,6 +97,7 @@ CONFIG_PINCTRL_STMFX=y
+ CONFIG_DM_PMIC=y
+ # CONFIG_SPL_PMIC_CHILDREN is not set
+ CONFIG_PMIC_STPMIC1=y
++CONFIG_DM_REGULATOR=y
+ CONFIG_DM_REGULATOR_FIXED=y
+ CONFIG_DM_REGULATOR_GPIO=y
+ CONFIG_DM_REGULATOR_STM32_VREFBUF=y
+-- 
+2.17.1
```
Comment (texierp) [inline recipes-bsp/u-boot/u-boot-mainline/0001-configs-stm32mp1-Remove-ADC.patch]: Generated with savedefconfig ?

**Review by texierp — CHANGES_REQUESTED** [review]

File: recipes-bsp/u-boot/u-boot-mainline/0001-configs-stm32mp1-Remove-ADC.patch (hunk 2 above)
Comment (jorisoffouga) [reply recipes-bsp/u-boot/u-boot-mainline/0001-configs-stm32mp1-Remove-ADC.patch]: Yes

File: recipes-bsp/u-boot/u-boot-stm-common.inc (hunk 1 above)
Comment (jorisoffouga) [reply recipes-bsp/u-boot/u-boot-stm-common.inc]: Fixed

**Review by texierp — APPROVED** [review]

**Comment by texierp** [comment]
```
ERROR: u-boot-mainline-v2019.10+gitAUTOINC+8c56ea5c1e-r0 do_patch: Command Error: 'quilt --quiltrc /home/pjtexier/Bureau/Yocto/warrior/st-build/tmp/work/stm32mp157a_dk1-poky-linux-gnueabi/u-boot-mainline/v2019.10+gitAUTOINC+8c56ea5c1e-r0/recipe-sysroot-native/etc/quiltrc push' exited with 0  Output:
Applying patch 0001-configs-stm32mp1-Remove-ADC.patch
patching file configs/stm32mp15_basic_defconfig
Hunk #1 FAILED at 7.
Hunk #2 succeeded at 26 (offset -1 lines).
Hunk #3 FAILED at 58.
Hunk #4 succeeded at 99 (offset 1 line).
2 out of 4 hunks FAILED -- rejects in file configs/stm32mp15_basic_defconfig
Patch 0001-configs-stm32mp1-Remove-ADC.patch does not apply (enforce with -f)
```

File: recipes-bsp/u-boot/u-boot-stm-common.inc [hunk 3]
```diff
@@ -8,7 +8,7 @@ DEPENDS += "bison-native"
 
 SRC_URI = " \
 	git://gitlab.denx.de/u-boot/u-boot.git;protocol=https \
-	file://0001-remove-usb-power.patch \
+	file://0001-configs-stm32mp1-Remove-ADC.patch \
 	file://0001-stm32mp1-Replace-ENV_IS_EXT4-to-MMC.patch \
 	file://0002-stm32mp1-Enable-redundant-environment.patch \
 "
```
Comment (texierp) [inline recipes-bsp/u-boot/u-boot-stm-common.inc:14]:
```
ERROR: u-boot-mainline-v2019.10+gitAUTOINC+8c56ea5c1e-r0 do_patch: Command Error: 'quilt --quiltrc /home/pjtexier/Bureau/Yocto/warrior/st-build/tmp/work/stm32mp157a_dk1-poky-linux-gnueabi/u-boot-mainline/v2019.10+gitAUTOINC+8c56ea5c1e-r0/recipe-sysroot-native/etc/quiltrc push' exited with 0  Output:
Applying patch 0001-configs-stm32mp1-Remove-ADC.patch
patching file configs/stm32mp15_basic_defconfig
Hunk #1 FAILED at 7.
Hunk #2 succeeded at 26 (offset -1 lines).
Hunk #3 FAILED at 58.
Hunk #4 succeeded at 99 (offset 1 line).
2 out of 4 hunks FAILED -- rejects in file configs/stm32mp15_basic_defconfig
Patch 0001-configs-stm32mp1-Remove-ADC.patch does not apply (enforce with -f)
```

**Review by texierp — CHANGES_REQUESTED** [review]

**Review by texierp — APPROVED** [review]

**Pull request merged**